stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
-from _typeshed import Incomplete
+from collections.abc import Callable, Generator, Iterator, Sequence
+from re import Match, Pattern
+from typing import Any, Generic, TypeVar, overload
+from typing_extensions import Self, TypeAlias
 
-def __getattr__(name: str) -> Incomplete: ...
+_TransitionResult: TypeAlias = tuple[list[str], str | None, list[str]]
```

Comment:
I think it should be fine to have an alias for the StateName so that we know that `str | None` is a state name or None.

Reply:
do you mean this?

```python
_StateName: TypeAlias = str | None
_TransitionResult: TypeAlias = tuple[list[str], _StateName, list[str]]
_TransitionMethod: TypeAlias = Callable[[Match[str], list[str], str], _TransitionResult]
```

Reply:
Ah yes, it might be redundant but it might help people I think. At least you know what this str is